src/main/java/io/cdap/e2e/utils/CdfHelper.java

```diff
@@ -29,15 +43,96 @@
     SeleniumDriver.openPage(SeleniumHelper.readParameters("CDFURL"));
     try {
       SeleniumDriver.getDriver().switchTo().alert().accept();
       SeleniumDriver.waitForPageToLoad();
     } catch (NoAlertPresentException Ex) {
       SeleniumDriver.waitForPageToLoad();
     }
+  }
+
+  static int getCountOfNoOfRecordsTransferredToBigQueryIn(String arg0) throws IOException, InterruptedException {
+    int countRecords;
+    countRecords = GcpClient.countBqQuery(SeleniumHelper.readParameters(arg0));
+    BeforeActions.scenario.write("**********No of Records Transferred in table" + SeleniumHelper.readParameters(arg0)
+                                   + "*:"
+                                   + countRecords);
+    Assert.assertTrue(countRecords > 0);
+    return countRecords;
+  }
+
+  static void selectSourceGCS() throws InterruptedException {
+    CdfStudioActions.selectGCS();
+  }
+
+  default void gcsProperties(String formatType) throws IOException, InterruptedException {
+    CdfGcsActions.enterReferenceName();
+    CdfGcsActions.enterProjectId();
+    CdfGcsActions.selectFormat(formatType);
+    CdfGcsActions.skipHeader();
+    CdfGcsActions.getSchema();
+    SeleniumHelper.waitElementIsVisible(CdfGCSLocators.getSchemaButton);
+  }
+
+  default void saveAndDeployPipeline() throws InterruptedException {
+    CdfStudioActions.pipelineName();
+    CdfStudioActions.pipelineNameIp("Pipeline-" + UUID.randomUUID().toString());
+    CdfStudioActions.pipelineSave();
+    SeleniumHelper.waitElementIsVisible(CdfPipelineRunLocators.savedSuccessMessage, 1);
+    CdfStudioActions.pipelineDeploy();
+  }
+
+  default void runThePipelineInRuntime() throws InterruptedException {
+    CdfPipelineRunAction.runClick();
+  }
+
+  default void waitTillPipelineToComplete() throws InterruptedException {
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 600000);
+    wait.until(ExpectedConditions.or(ExpectedConditions.visibilityOfElementLocated(
+      By.xpath("//*[@data-cy='Succeeded']")), ExpectedConditions.visibilityOfElementLocated(
+      By.xpath("//*[@data-cy='Failed']"))));
+  }
+
+  default void verifyThePipelineStatusIsForTheCurrentPipeline(String status) throws InterruptedException {
+    Assert.assertTrue(SeleniumDriver.getDriver().findElement(By.xpath("//*[@data-cy='" + status + "']")).isDisplayed());
+  }
+
+  default void captureLogs() throws FileNotFoundException {
+    CdfPipelineRunAction.logsClick();
+    BeforeActions.scenario.write(CdfPipelineRunAction.captureRawLogs());
+  }
+
+  default void succeedPipelineVerification() {
+    CdfLogActions.validateSucceeded();
+  }
+
+  default int recordIn() {
+    return CdfPipelineRunAction.getRecordInCount();
+  }
+
+  default int recordOut() {
+    return CdfPipelineRunAction.getRecordOutCount();
+  }
+
+  default void deleteTable(String table) throws IOException, InterruptedException {
```

Comment:
should be deleteBigQueryTable(). This is a general framework, so don't assume we're always dealing with BQ/GCS

Reply:
done